halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Node.java

```diff
@@ -53,6 +54,9 @@
 @Slf4j
 public abstract class Node implements Validatable {
 
+  // @LocalFile fields whose file path is relative
+  @JsonIgnore private Map<Field, String> relativeFileReferences = new ConcurrentHashMap<>();
```

Comment:
We should keep a reference by Node + Field - not just Field, otherwise a Node using more than one relative path won't be tracked/restored.

Reply:
The map is not globally unique, there's a map for each instance of `Node`, and `Field` is already unique in a given node hierarchy. I did a test with two kubernetes accounts pointing to relative kubeconfig files (same `Field` definition, two different node instances), and both were correctly generated in `staging` directory.

Reply:
Ah good point, missed that.